test/unit/publishProjectWebViewController.test.ts

```diff
@@ -659,4 +661,107 @@ suite("PublishProjectWebViewController Tests", () => {
         ).to.be.true;
     });
     //#endregion
+
+    //#region Generate Script Tests
+    test("generatePublishScript reducer closes dialog and triggers script generation", async () => {
+        const controller = createTestController();
+        await controller.initialized.promise;
+
+        // Set up state with all required fields for script generation
+        controller.state.formState.serverName = "localhost";
+        controller.state.formState.databaseName = "TestDatabase";
+        controller["_connectionUri"] = "mssql://test-connection-uri";
+        controller.state.projectFilePath = "c:/work/TestProject.sqlproj";
+
+        // Mock the panel dispose method using sandbox stub
+        const panelDisposeSpy = sandbox.stub();
+        Object.defineProperty(controller, "panel", {
+            value: { dispose: panelDisposeSpy },
+            writable: true,
+            configurable: true,
+        });
+
+        // Spy on executePublishAndGenerateScript to verify it's called
+        const executePublishSpy = sandbox.stub(
+            controller as typeof controller & {
+                executePublishAndGenerateScript: (state: unknown, isPublish: boolean) => void;
+            },
+            "executePublishAndGenerateScript",
+        );
+
+        const reducerHandlers = controller["_reducerHandlers"] as Map<string, Function>;
+        const generatePublishScript = reducerHandlers.get("generatePublishScript");
+        expect(generatePublishScript, "generatePublishScript reducer should be registered").to
+            .exist;
+
+        // Invoke the reducer
+        await generatePublishScript(controller.state, {});
+
+        // Verify dialog was closed
+        expect(
```

Comment:
In this case, you can use sinon-chat (see adapter.test.ts) to do something like:

`expect(panelDisposeSpy).to.have.been.calledOnce()`